slides/src/content/routing.md

```diff
@@ -189,109 +217,247 @@ export class ProductDetails {
+
+import { LazyComponent } from './lazy.component';
+
+const routes: Routes = [
+  { path: '', component: LazyComponent }
+];
+
+export const routing: ModuleWithProviders = RouterModule.forChild(routes);
 ```
 
-[View Example](https://plnkr.co/edit/FOeNsspRfjg3PVLiUsip?p=preview)
+Our `LazyComponent` has now been setup for lazy loading. If we start the app, we will see that the `LazyComponent` does not get loaded right away. It only gets loaded the first time, when we navigate to the `lazy` route. If we navigate back and forth now between an eager loaded and lazy loaded module, we will see that both components are cached now. 
+
+[View Example](https://plnkr.co/edit/vpCqRHDAj7V6mlN1AknN?p=preview)
 
 ---
 
-## Child Routes (3/3)
-To access a parent's route parameters:
+## Route Authorization (1/3)
 
-```javascript
-export default class Overview {
-  constructor(private router: Router,
-    private route: ActivatedRoute) {}
+To control whether the user can navigate to or away from a given route, we can use route guards.
 
-  ngOnInit() {
-    // Get parent ActivatedRoute of this route.
-    this.sub = this.router.routerState.parent(this.route)
-      .params.subscribe(params => {
-        // params['id'] is parent's product id
-      });
+In order to use route guards, we must register them with the specific routes we want them to run for.
+
+```ts
+const routes: Routes = [
+  { path: 'home', component: HomePage },
+  {
+    path: 'accounts',
+    component: AccountPage,
+    canActivate: [LoginRouteGuard],
+    canDeactivate: [SaveFormsGuard]
+  }
+];
+```
+
+---
+
+## Route Authorization (2/3)
+
+To guard a route against unauthorized activation, we can implement the `CanActivate` interface by implementing the `canActivate` function.
+
+When `canActivate` returns `true`, the user can activate the route. When `canActivate` returns `false`, the user cannot access the route.
+
+```ts
+@Injectable()
+export class LoginRouteGuard implements CanActivate {
+
+  constructor(private loginService: LoginService) {}
+
+  canActivate() {
+    return this.loginService.isLoggedIn(); // true or false
   }
 }
 ```
 
 ---
 
-## Passing Optional Parameters (1/2)
+## Route Authorization (3/3)
 
-Use the `[queryParams]` directive along with `[routerLink]` to pass query parameters. For example:
+CanDeactivate works in a similar way to CanActivate but there are some important differences.
 
-```html
-<a [routerLink]="['product-list']" [queryParams]="{ page: 99 }">Go to Page 99</a>
+The canDeactivate function passes the component being deactivated as an argument to the function.
+
+We can use that component to determine whether the user can deactivate.
+
+```ts
+canDeactivate(component: AccountPage) {
+  return component.areFormsSaved();
+}
 ```
 
-Alternatively, we can navigate programmatically using the `Router` service:
+---
 
-```javascript
-  goToPage(pageNum) {
-    this.router.navigate(['/product-list'], { queryParams: { page: pageNum } });
+## Child Routes (1/4)
```

Comment:
To me, this would flow better if it were earlier. Maybe right after route parameters and before lazy loading?